Azure/azure-cli
# [Network] Support private link for managed disks

**Backgroud** 

Feature Request: CLI Support for private links of managed disks #13917 

This PR is a part of supporting private links for managed disks. Private links for managed disks is a little different from the already supported private links in CLI (e.g, keyvault). It use the disk-access as an intermediate resource rather than connect with the target resource directly. Therefore, to support the private links for managed disks, we have to do the following things. This PR works to support the third step. 

> 1. Support CURD of the new resource disk-access
> 2. Support create/update a disk/snapshot to associate it a disk-access resource
> 3. **Support private links for managed disks**

**Description**  


This PR provides support for:
1. `az network private-link-resource list` for --type Microsoft.Compute/diskAccesses
2. `az network private-endpoint-connection list` for --type Microsoft.Compute/diskAccesses

As service currently does not support for manual approval of the connection, and does not provide the APIs, these commands are not supported.
1. `az network private-endpoint-connection approve`
2. `az network private-endpoint-connection reject`
3. `az network private-endpoint-connection delete`
4. `az network private-endpoint-connection show`

We will raise the following error if users are trying to use these commands.
```
Resource provider Microsoft.Compute/diskAccesses currently does not support this operation
```
**Testing Guide**  

1. Create a disk-access and check private link resource
```
az disk-access create -g {rg} -l {loc} -n {disk_access}

az network private-link-resource list -g {rg} -n {disk_access} --type Microsoft.Compute/diskAccesses
```
2. Prepare and enable VNet
```
az network vnet create -g {rg} -n {pe_vnet} --subnet-name {pe_subnet}

az network vnet subnet update -g {rg} -n {pe_subnet} --vnet-name {pe_vnet}  
   --disable-private-endpoint-network-policies true
```
3. Create a private endpoint connection and check the auto-approval
```
az network private-endpoint create -g {rg} -n {pe_name} --vnet-name {pe_vnet}
  --subnet {pe_subnet} --private-connection-resource-id {disk_access_id}
  --group-ids disks --connection-name {pe_connection}

az network private-endpoint-connection list -g {rg} -n {disk_access} --type Microsoft.Compute/diskAccesses
```
4. Create a disk or snapshot and associate it with the disk-access to test private connection
